pkg/hpa/service.go

```diff
@@ -50,9 +51,10 @@ func New(
 	maximumTargetResourceUtilization,
 	tortoiseHPATargetUtilizationMaxIncrease int,
 	tortoiseHPATargetUtilizationUpdateInterval time.Duration,
-	maximumMinReplica, maximumMaxReplica int32,
+	maximumMinReplica,
 	minimumMinReplicas int32,
 	externalMetricExclusionRegex string,
+	config *config.Config,
```

Comment:
Please avoid passing the entire config. It would lose the visibility what parameters of config this function uses. (you would have to check the function implementation)